examples/hnas/example_mlp.py

```diff
@@ -0,0 +1,228 @@
+    args = parse_args()
+
+    logging.root.setLevel(logging.INFO)
+
+    # In case you want to run AutoGluon across multiple instances, you need to provide it with a list of
+    # the IP addresses of the instances. Here, we assume that the IP addresses are stored in a .yaml file.
+    # If you want to run AutoGluon on a single instance just pass None.
+    # However, keep in mind that it will still parallelize the optimization process across multiple threads then.
+    # If you really want to run it purely sequentially, set the num_cpus equal to the number of VCPUs of the machine.
+    dist_ip_addrs = yaml.load(args.hostfile) if args.hostfile is not None else []
+
+    del args.hostfile
+
+    print("Got worker host IP addresses [{}]".format(dist_ip_addrs))
+
+    if args.scheduler == "hyperband_stopping":
+        hyperband_type = "stopping"
+
+    elif args.scheduler == "hyperband_promotion":
+        hyperband_type = "promotion"
+
+    brackets = 1  # setting the number of brackets to 1 means that we run effectively successive halving
+
+    scheduler = ag.scheduler.HyperbandScheduler(objective_function,
+                                                resource={'num_cpus': args.num_cpus, 'num_gpus': args.num_gpus},
+                                                # Autogluon runs until it either reaches num_trials or time_out
+                                                num_trials=args.num_trials,
+                                                time_out=args.timeout,
+                                                # This argument defines the metric that will be maximized.
+                                                # Make sure that you report this back in the objective function.
+                                                reward_attr='performance',
+                                                # The metric along we make scheduling decision. Needs to be also
+                                                # reported back to AutoGluon in the objective function.
+                                                time_attr='epoch',
+                                                brackets=brackets,
+                                                checkpoint=None,
+                                                searcher="bayesopt",  # Defines searcher for new configurations
+                                                dist_ip_addrs=dist_ip_addrs,
+                                                training_history_callback=callback,
+                                                training_history_callback_delta_secs=args.store_results_period,
+                                                reduction_factor=3,
+                                                type=hyperband_type,
+                                                # defines the minimum resource level for Hyperband,
+                                                # i.e the minimum number of epochs
+                                                grace_period=1
```

Comment:
User argument?

Reply:
I am not exactly sure what you mean here

Reply:
I mean we could make grace_period and reduction_factor arguments in the CommandParser.

Reply:
makes sense, I've added that